app/domain/batch_process.rb

```diff
@@ -0,0 +1,17 @@
+class BatchProcess
+  attr_reader :output_documents
+
+  def initialize(output_documents)
+    @output_documents = output_documents
+  end
+
+  def csv
+    [headers].concat(output_documents.map(&:csv)).join
+  end
+
+  private
+
+  def headers
+    OutputDocument::CSVRenderer::FIELDS.join('|') + "\n"
```

Comment:
I had thought the headers would need some sort of special treatment but now that you mention it I think treating them just like another row would be the thing to do per [CSVRenderer#render](https://github.com/guidance-guarantee-programme/output/blob/master/app/domain/output_document/csv_renderer.rb#L20-L22)